rviz_plugins/eStop_plugin/src/bare_panel.h

```diff
@@ -0,0 +1,37 @@
+#ifndef BARE_PANEL_H
+#define BARE_PANEL_H
+
+#include <ros/ros.h>
+#include <QLabel>
+#include <std_msgs/Bool.h>
+
+#include <rviz/panel.h>
+
+class QLineEdit;
+
+namespace bare
+{
+
+  class BarePanel: public rviz::Panel
+  {
+    Q_OBJECT
+    public:
+      BarePanel( QWidget* parent = 0 );
+
+    private:
+      void subCallback(const std_msgs::Bool &msg);
+
+    public Q_SLOTS:
+
```

Comment:
Q_SLOTS are actually necessary for this because the qt GUI objects communicate with Q_SLOTS.